sdk/formrecognizer/Azure.AI.FormRecognizer/src/AnalyzeResult.cs

```diff
@@ -11,7 +12,63 @@ namespace Azure.AI.FormRecognizer.DocumentAnalysis
     [CodeGenModel("AnalyzeResult")]
     public partial class AnalyzeResult
     {
+        internal AnalyzeResult(ApiVersion apiVersion, string modelId, StringIndexType stringIndexType, string content, IReadOnlyList<DocumentPage> pages, IReadOnlyList<DocumentTable> tables, IReadOnlyList<DocumentKeyValuePair> keyValuePairs, IReadOnlyList<DocumentEntity> entities, IReadOnlyList<DocumentStyle> styles, IReadOnlyList<AnalyzedDocument> documents)
+        {
+            ApiVersion = apiVersion;
+            ModelId = modelId;
+            StringIndexType = stringIndexType;
+            Content = content;
+            Pages = pages;
+            Tables = tables;
+            KeyValuePairs = keyValuePairs;
+            Entities = entities;
+            Styles = styles;
+            Documents = documents;
+
+            // Set page references for the 'Words' convenience properties.
+
+            foreach (DocumentPage page in Pages)
+            {
+                foreach (DocumentLine line in page.Lines)
+                {
+                    line.Page = page;
+                }
+            }
```

Comment:
We need to access the `DocumentWords` from multiple other document elements, so we need to pass some references around. I'm adding an internal field to these elements (in this case, `DocumentLine.Page`) so these can be used to access the word references when `Words` is called.

I decided to put all the reference passing in the same place (so the logic doesn't get scattered around).